rust/arrow/src/compute/kernels/comparison.rs

```diff
@@ -66,6 +66,31 @@ macro_rules! compare_op {
     }};
 }
 
+macro_rules! compare_op_scalar {
+    ($left: expr, $right:expr, $op:expr) => {{
+        let null_bit_buffer =
```

Comment:
since you're comparing to a non-null scalar, you shouldn't need to compute a bitmap, you could use `$left`'s null buffer: `$left.data().null_buffer()`

Reply:
thanks @nevi-me , you are right, I have changed the scalar comparison code to clone the left null buffer